component/metrics/remotewrite/remote_write.go

```diff
@@ -173,16 +173,11 @@ func (c *Component) Update(newConfig component.Arguments) error {
 func (c *Component) Receive(ts int64, metricArr []*metrics.FlowMetric) {
 	app := c.storage.Appender(context.Background())
 	for _, m := range metricArr {
-		// TODO this should all be simplified into one call
-		if m.GlobalRefID == 0 {
-			globalID := metrics.GlobalRefMapping.GetOrAddGlobalRefID(m.Labels)
-			m.GlobalRefID = globalID
-		}
-		localID := metrics.GlobalRefMapping.GetLocalRefID(c.opts.ID, m.GlobalRefID)
-		newLocal, err := app.Append(storage.SeriesRef(localID), m.Labels, ts, m.Value)
+		localID := metrics.GlobalRefMapping.GetLocalRefID(c.opts.ID, m.GlobalRefID())
+		newLocal, err := app.Append(storage.SeriesRef(localID), m.LabelsCopy(), ts, m.Value())
```

Comment:
I'm still concerned about this making a copy of the labels for every single sample.

Reply:
If we ever expose relabel_configs for remote_write then they could mutate the labels.